### PR TITLE
Local media files caching

### PR DESCRIPTION
**Pull request by eddyStreamlabs**

### CODE REVIEW

File: deps/media-playback/media-playback/media.c [hunk 1]
```diff
@@ -258,125 +258,183 @@ static inline bool mp_media_can_play_frame(mp_media_t *m,
...
+
+		audio->samples_per_sec = f->sample_rate * m->speed / 100;
+		audio->speakers = convert_speaker_layout(f->channels);
+		audio->format = convert_sample_format(f->format);
+		audio->frames = f->nb_samples;
+		audio->timestamp = m->base_ts + d->frame_pts - m->start_ts +
+			m->play_sys_ts - base_sys_ts;
+
+		if (audio->format == AUDIO_FORMAT_UNKNOWN)
+			return;
```
Comment (summeroff) [inline deps/media-playback/media-playback/media.c]: "audio" and "audio->data[]" can be leaked here?

Comment (eddyStreamlabs) [reply deps/media-playback/media-playback/media.c]: No they should be correctly released in the `clear_cache` method

Comment (summeroff) [reply deps/media-playback/media-playback/media.c]: to be removed in clear_cache they have to be saved "m->audio". 
But if function exit on "if (audio->format == AUDIO_FORMAT_UNKNOWN) return ;" then "audio" doesn't have a chance to be saved to it.

Comment (eddyStreamlabs) [reply deps/media-playback/media-playback/media.c]: should be fixed

**Pull request merged**